sympy/printing/str.py

```diff
@@ -74,12 +74,15 @@ def _print_BooleanTrue(self, expr):
     def _print_BooleanFalse(self, expr):
         return "False"
 
+	def _print_Not(self, expr):
+		return '~\(' + expr.args[0] + '\)' 
```

Comment:
maybe expressions should be parenthesized based on the node precedence.

Try using 
https://github.com/SagarB-97/sympy/blob/9f3f6b99c729d243060c4f5f0f49b9452bfbc972/sympy/printing/str.py#L27